### ISSUE TITLE
Add extrude support, use extrude tool by default for rectangular objects, add toggle for it in UI

### ISSUE THREAD

**Issue by d1g** [openstreetmap/iD#2986]
It makes sense to enable this mode when editing objects with building=*, but not natural=water (and such).

https://josm.openstreetmap.de/wiki/Help/Action/Extrude
Not everything must be implemented: "ctrl" mode (as default way to _move_ way segments) and probably "basic mode"

It makes no sense to draw arbitrary shaped object, then use Q tool, then correct some portion of the object, then press Q again.

**Comment by bhousel** [comment]
Yes I would also like the ability to select segments and extrude them.

Both extrusion (selecting a segment and moving it), and polygon operations (#435), are useful tools described here:  https://www.mapbox.com/blog/fast-building-tracing/

**Comment by hb0nd** [comment]
I'd _really_ love to be able to extrude, it's the main thing keeping me mapping buildings in JOSM.
Extruding speeds up rectangular building creation so much, I feel it'd be a very valuable feature to add to iD ^-^
